apm-agent-plugins/apm-api-plugin/src/test/java/co/elastic/apm/agent/pluginapi/SpanInstrumentationTest.java

```diff
@@ -156,6 +162,156 @@ void testTraceHeaders() {
         span.end();
     }
 
+    @Test
+    void testSetDestinationAddressWithNonNullValues() {
+        Span span = transaction.startSpan("foo", "subtype", "action")
+            .setDestinationAddress("address", 80);
+        endSpan(span);
+
+        assertThat(reporter.getSpans().size()).isEqualTo(1);
+        assertDestination("address", 80);
+    }
+
+    @Test
+    void testSetDestinationAddressWithNegativePort() {
+        Span span = transaction.startSpan("foo", "subtype", "action")
+            .setDestinationAddress("address", -1);
+        endSpan(span);
+
+        assertThat(reporter.getSpans().size()).isEqualTo(1);
+        assertDestination("", 0);
+    }
+
+    @Test
+    void testSetDestinationAddressWithNullAddress() {
+        Span span = transaction.startSpan("foo", "subtype", "action")
+            .setDestinationAddress(null, 80);
+        endSpan(span);
+
+        assertThat(reporter.getSpans().size()).isEqualTo(1);
+        assertDestination("", 0);
+    }
```

Comment:
Is there a requirement that a valid port only comes with a valid address and vice versa? If not, please change the logic and tests so that a user can provide only one of them valid.

Reply:
I didn't find a requirement. Changed logic.